### ISSUE TITLE
create a new Tab which can list the graphs having Nan

### ISSUE THREAD

**Issue by gj00354347** [Cacti/cacti#5357]
## Problem

We use Cacti 1.2.18 in production to monitor approximately 5,000 devices and 500,000 graphs. Clients regularly report graphs showing `NaN`, for reasons including removed interfaces, SNMP timeouts, or values exceeding configured limits.

## Requested solution

Add a **NaN Graphs** view with device and date/duration filters so operators can find affected graphs efficiently.

Alternatively, provide an efficient, low-I/O and low-memory script that can run weekly or monthly and return a list of graphs containing `NaN` values.

**Comment by anarkia1976** [comment]
> Please have a look at #4943

hi @xmacan,
Good, but how we can find graphs with NAN values?
could you direct on how to identify by red check?
thanks a lot.

**Comment by gj00354347** [comment]
@xmacan Hey many thanks ); for pointing me to this .

just a few questions , As I am on 1.2.18 and do not know the GUI of 1.2.23 where this feature is available .
so my question is in the ticket , I see stale rrd checks , there I see 3 columns Date , DS and Message , so does the message field contain the Graph name that's in Cacti GUI or is there any other column ?? 

because Knowing DS is good but we need to map those DSes with Graph Name , so is that available ?? and can we put some kind of alerting in this through Thold so that we can generate ticket on this and can start working on this rather than manually checking the page every 2 or 3 weeks.

**Comment by netniV** [comment]
If you look at the 1.2.x code, take a look at the clog_regex_xxx functions, these convert various IDs to names.

If you look at the 1.3.x code, this has been moved from clog to text_regex_xxx to make them more generally available.

**Comment by gj00354347** [comment]
@netniV  Hi Vinten ,

thanks . so is that clog_regex_xx feature available in GUI as well or I will have to make use of it through back end CLI ??
if this is available in GUI of cacti , it will make our life much easier .

**Comment by xmacan** [comment]
> @xmacan Hey many thanks ); for pointing me to this .
> 
> just a few questions , As I am on 1.2.18 and do not know the GUI of 1.2.23 where this feature is available . so my question is in the ticket , I see stale rrd checks , there I see 3 columns Date , DS and Message , so does the message field contain the Graph name that's in Cacti GUI or is there any other column ??
> 
> because Knowing DS is good but we need to map those DSes with Graph Name , so is that available ?? and can we put some kind of alerting in this through Thold so that we can generate ticket on this and can start working on this rather than manually checking the page every 2 or 3 weeks.

I will look at it. I think that add graph names is small change

**Comment by gj00354347** [comment]
Many Thanks @xmacan , eagerly awaiting  this feature  in cacti 1.2.25 );

**Comment by netniV** [comment]
The regex code is already present in the code, but may not be utilised elsewhere.  With 1.3.x, the code was moved and made to work with and without providing links to the HTML pages to manage them, so any changes in 1.2.x will require porting manually to 1.3.x

**Comment by TheWitness** [comment]
We have to be careful with Regex due to a recent CVE.  We've put some mitigation in place.  Hopefully it's good enough.

**Comment by TheWitness** [comment]
How many NaN's means NaN?  Anyway, I think it's almost there already from my perspective.

**Comment by anarkia1976** [comment]
> ![Screenshot 2023-06-12 at 17-18-17 Console Utilities RRD Check](https://user-images.githubusercontent.com/26485719/245189767-e0c2a612-a2d3-4bba-a5f0-e72aa3631303.png)
> 
> @gj00354347 , @anarkia1976 - something like this?

good very good, pefect.

**Comment by netniV** [comment]
Just remember, if this is a new feature, it's a 1.3.x implementation. :)

**Comment by xmacan** [comment]
damn, I have prepared it for 1.2.x in last few minutes

**Comment by gj00354347** [comment]
> ![Screenshot 2023-06-12 at 17-18-17 Console Utilities RRD Check](https://user-images.githubusercontent.com/26485719/245189767-e0c2a612-a2d3-4bba-a5f0-e72aa3631303.png)
> 
> @gj00354347 , @anarkia1976 - something like this?

Yes very much like this , where we have filter like time span like nan for 2 hours or nan for 2 days or nan for 2 months .

**Comment by gj00354347** [comment]
> I think about email notification. RRDchecker is started periodically. I would like to send email when periodical check found any issue.

Yes very much ok approach , like run that check every week or every 2 weeks . But one thing we can improve here like in addition to sending emails , can we send log like thold module which sends the breach log to /var/log/err and a ticket is generated in ticketing tool . Here we can set a filter like if x, y or x device is having problem then send the msg to err log or if the No of graphs having nan is more than 10 or 15 then send a consolidated log which lists the problematic interfaces or unpopulated interfaces in err log and then ticketing tool will come in picture.

**Comment by xmacan** [comment]
The result = I will prepare it for 1.3.x cccording to the notes and requirements in #5373

**Comment by somethingwithproof** [comment]
No PR yet for NaN graph listing tab. Second most-discussed open enhancement (#18 comments). PRs welcome.

**Comment by xmacan** [comment]
@somethingwithproof - This has already been partially resolved. It doesn't directly address NaN values, but we log polling errors to the `host_errors` table. In the console -> Administration -> Data Sources -> check the "Errored" checkbox.
The Cacti statistics template now includes a new graph titled "Device - Polling Errors" (per device) and a "Total Errors" graph.

**Comment by somethingwithproof** [comment]
Thanks for the pointer. The polling-errors graphs are a genuinely useful related signal, but they track poll failures/error counts, not NaN values in otherwise-successful polls, so they don't cover the original ask (a tab listing graphs with NaN data points). Leaving this open as the NaN-listing tab is still unaddressed.
